### PR TITLE
Support external Postgresql database server

### PR DESCRIPTION
**Pull request by azhelev**
## Description of the change

Adds support for using external Postgresql server

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by nauxliu** [comment]
Will there be a problem if multiple workers share one Postgresql instance? as far as I know every worker has its own Postgresql.

**Review by Dr4il — APPROVED** [review]
Nice work

File: templates/statefulset.yaml [hunk 1]
```diff
@@ -81,21 +81,49 @@ spec:
           - name: DEST_TRANSFORM_URL
             value: "http://{{ include "transformer.fullname" . }}:{{ .Values.transformer.service.port}}"
           - name: COMPUTE_DB_HOST_IN_K8S
+            {{- if .Values.external_postgresql.enabled }}
```
Comment (snarkychef) [inline templates/statefulset.yaml:84]: Having two boolean fields viz. external_postgresql.enabled and postgresql.enabled can cause consistency issues. Does it make sense to use just postgresql.enabled?

File: templates/statefulset.yaml [hunk 2]
```diff
@@ -81,21 +81,49 @@ spec:
...
+            {{- end }}
           - name: JOBS_DB_USER
+            {{- if .Values.external_postgresql.enabled }}
+            value: "{{ .Values.external_postgresql.postgresqlUsername }}"
+            {{ else }}
             value: "{{ .Values.postgresql.postgresqlUsername }}"
+            {{- end }}
           - name: JOBS_DB_PORT
+            {{- if .Values.external_postgresql.enabled }}
+            value: "{{ .Values.external_postgresql.postgresqlPort }}"
```
Comment (snarkychef) [inline templates/statefulset.yaml:107]: Add default port as 5432?

File: templates/statefulset.yaml [hunk 3]
```diff
@@ -81,21 +81,49 @@ spec:
...
+            {{- end }}
           - name: JOBS_DB_PASSWORD
+            {{- if .Values.external_postgresql.enabled }}
+            value: "{{ .Values.external_postgresql.postgresqlPassword }}"
+            {{ else }}
             value: "{{ .Values.postgresql.postgresqlPassword }}"
+            {{- end }}
+          {{- if .Values.external_postgresql.enabled }}
+          - name: JOBS_DB_SSL_MODE
+            value: "{{ .Values.external_postgresql.postgresqlSslMode }}"
```
Comment (snarkychef) [inline templates/statefulset.yaml:125]: Add default value `disable` ?

File: values.yaml [hunk 4]
```diff
@@ -132,6 +133,19 @@ postgresql:
     limits:
       memory: 4096Mi
 
+# Use this if you have external Postgresql database
+# in this case you should set postgresql.enabled to false above and enabled to true here
+# the postgresqlUsername role must have CREATEDB privilege
+# the postgresqlDatabase must exists and be owned by the postgresqlUsername role
+external_postgresql:
+  enabled: false
+  postgresqlHost: example.server
```
Comment (snarkychef) [inline values.yaml:142]: comment out this block? and keep values only for variables which have default values defined in code.

File: Chart.yaml [hunk 5]
```diff
@@ -32,3 +32,9 @@ appVersion: 1.16.0
 #    import-values: # (optional)
 #      - ImportValues holds the mapping of source values to parent key to be imported. Each item can be a string or pair of child/parent sublist items.
 #    alias: (optional) Alias usable alias to be used for the chart. Useful when you have to add the same chart multiple times
+
+dependencies:
+- condition: postgresql.enabled
+  name: postgresql
+  repository: https://charts.bitnami.com/bitnami
```
Comment (snarkychef) [inline Chart.yaml:39]: instead of helm repository, can we use path based chart dependency as we already have code in this repo.

**Review by snarkychef — CHANGES_REQUESTED** [review]

**Comment by ldaugusto** [comment]
> Will there be a problem if multiple workers share one Postgresql instance? as far as I know every worker has its own Postgresql.

That's a good point. If a single db instance can handle multiple servers, why does the chart load a new postgres for every server instance.

**Comment by gitcommitshow** [comment]
@azhelev can you please check the comments in this PR

**Comment by gitcommitshow** [comment]
As shared by @psrikanth88, this PR has some unaddressed comments. And it would work with only one replica of statefulset. If there are multiple, it needs extra configuration support in our code to use different schema in same external postgres for each replica.

**Comment by azhelev** [comment]
I don't use this anymore, i have no interest in completing this PR. If someone wants to take it up feel free.

**Pull request closed without merging**